### DISCUSSION TITLE
LINQ None method (aka !Any)

### DISCUSSION THREAD

**Discussion by maxkoshevoi** [dotnet/runtime#54743] [Ideas] [answered]
I'm sure, this question was already raised here, but I couldn't find it.
What do you think about adding `None` method to LINQ to complement `Any`?

Main reason is that I see good amount of code that checks for empty collections like this. Which is not very readable.
```cs
if (!someLongCollectionName.Any())
```

As for other benefits. Third condition reads much better:
```cs
!arr.Any(i => i == 5)
arr.All(i => i != 5)
arr.None(i => i == 5)
```

**Comment by stephentoub** [comment, marked as answer]
https://github.com/dotnet/runtime/issues/27661

**Reply by JansthcirlU** [reply]
I can see why you're hesitant to add `Some` and `OnlySome`, but `None` seems like a very natural addition to me.

**Reply by TonyValenti** [reply]
Im a big fan of .None. I already have that extension method in my standard library. Would love to see it become core.
